test/jdk/java/net/httpclient/HttpResponseInputStreamInterruptTest.java

```diff
@@ -162,10 +164,14 @@ static Thread createClientThread(CountDownLatch interruptReadyLatch, int port) {
                 assertEquals(firstOutput, FIRST_MESSAGE);
 
                 // countdown on latch, and assert that an IOException is throw due to the interrupt
+                // and assert that the cause is a InterruptedException
                 interruptReadyLatch.countDown();
-                assertThrows(IOException.class, () -> response.body().readAllBytes(), "expected IOException");
-            } catch (Exception e) {
-                throw new RuntimeException(e);
+                var thrown = assertThrows(IOException.class, () -> response.body().readAllBytes(), "expected IOException");
+                var cause = thrown.getCause();
+                assertTrue(cause instanceof InterruptedException, cause + " is not an InterruptedException");
```

Comment:
Ah - and of course the test should verify that the thread interrupted status is set too, since we specified that :-)

```
var thread = Thread.currentThread();
assertTrue(thread.isInterrupted(), "Thread " + thread + " is not interrupted");
```